packages/docusaurus-init/src/index.ts

```diff
@@ -64,6 +65,46 @@ function createTemplateChoices(templates: string[]) {
   ];
 }
 
+async function copyTemplate(
```

Comment:
I'd prefer to avoid handling edge cases in this file.

My suggestion here would be:
- Copy 100% of shared
- Copy 100% of the base template if any (only for classic-typescript)
- Copy 100% of the final template

In each case, you'd override existing files set by the previous steps. I don't think there is a need for some template to "remove" files that were copied from `shared` so it should work fine.

Reply:
I've removed the handling of `facebook`, but I still have to filter for `css` and `docusaurus.config.js` only, otherwise we will end up with `.js` files in the `typescript` template

Reply:
make sense thanks